### ISSUE TITLE
Overlap Helper Module

### ISSUE THREAD

**Issue by FarnazH** [theochem/iodata#21]
The `iodata.overlap_helper.py` contains a lot of pre-computed pure/cartesian transformation matrices generated by `horton.tools.harmonics.py`. The `iodata.overlap_helper.py` doesn't seem clear and nice, so I was wondering why we didn't include the functions generating them directly?

**Comment by matt-chan** [comment]
We should. I put the harmonics.py code into the gbasis repo I think, but I didn't think about the overlap code in IOData.

**Comment by FarnazH** [comment]
I have been working on `iodata` all day today trying to improve it, and I think its `overlap` code is over-complicated and even out of place to some extent. If I am not wrong, it's only used in the `molden` module to check normalization, so we don't even need the fastest overlap computation. I think we can write something much simpler for `iodata` and have a better/faster implementation in `gbasis`.

**Comment by matt-chan** [comment]
Yep, it's only used to check the normalization and normalize if needed. We don't need to have the full overlap matrix implementation in theory.

What did you have in mind?

**Comment by tovrstra** [comment]
You need the entire overlap matrix to check the normalization of *the orbitals*. We're not (only) checking the normalization of the basis functions.

A better name for this module would be good: `transform_cart_pure.py` or something along those lines.

We could compute the transformation matrices when needed and cached them. The `harmonics.py` in the `tools` directory of HORTON uses Sympy, which should be avoided. It is a tedious thing to implement in straight Python but good practice when your intuition for recursion relations is fading.

**Comment by PaulWAyers** [comment]
@FarnazH This seems to have been finished (a long time ago) but never closed.
